### ISSUE TITLE
Deal api crm.deals.searchApi.doSearch has an invalid sorts type

### ISSUE THREAD

**Issue by voho** [HubSpot/hubspot-api-nodejs#559]
The type of `sorts` is an array of strings; however, [according to your docs](https://developers.hubspot.com/beta-docs/guides/api/crm/search#sort-search-results), it should be also possible to specify sorting directions like this:

```js
    "sorts": [
      {
        "propertyName": "createdate",
        "direction": "DESCENDING"
      }
    ]
```

**Comment by kiransmith** [comment]
@voho I'm having the same problem - have you come across a working solution?

**Comment by nicolasmondain** [comment]
Just had a test on the `crm.contacts.searchApi.doSearch` (which works the exact same way):
- results are returned properly depending on the direction
```javascript
        const objectSearchRequest = {
          filterGroups: [{
            filters: [{ propertyName: 'firstname', operator: 'EQ', value: 'Nicolas' }],
          }],
          limit: 100,
          properties: ['firstname', 'lastname'],
          sorts: [{
            propertyName: 'lastname',
            direction: 'ASCENDING',
          }],
          after: 0,
        };
```
If you need your type definition to be up to date, I forked the repo a week ago and just applied an update that should resolve your issue. Going to open a new pull request.

```bash
> @hubspot/api-client@12.0.1 test
> ts-node node_modules/jasmine/bin/jasmine --config=jasmine.json

Randomized with seed 75149
Started
......................................................................


70 specs, 0 failures
Finished in 0.083 seconds
Randomized with seed 75149 (jasmine --random=true --seed=75149)
nicolasmondain@MBP-de-Nicolas hubspot-api-nodejs % npm run lint

> @hubspot/api-client@12.0.1 lint
> tslint -p tsconfig.json && npm run prettier:check


> @hubspot/api-client@12.0.1 prettier:check
> prettier --check "{,**/}*.{js,ts}"

Checking formatting...
All matched files use Prettier code style!
nicolasmondain@MBP-de-Nicolas hubspot-api-nodejs %
```
